include/sparrow/array/array_data_factory.hpp

```diff
@@ -39,6 +39,48 @@
 
 namespace sparrow
 {
+
+    // a range where the value type is convertible to bool
+    template<class BoolRange>
+    concept bool_convertible_range = std::ranges::range<BoolRange> &&
+        std::convertible_to<std::ranges::range_value_t<BoolRange>, bool>;
+
+    
+    // a range of nullable values / nullable<T>
+    template<class RangeOfNullables>
+    concept range_of_nullables = std::ranges::range<RangeOfNullables> && is_nullable<std::ranges::range_value_t<RangeOfNullables>>::value;
+        
+        
+
+    namespace detail
+    {
+
+        // a helper function to make the bitmap from a range
+        template<bool_convertible_range BoolRange>
+        array_data::bitmap_type make_array_data_bitmap(BoolRange&& range)
+        {
+            array_data::bitmap_type bitmap(std::ranges::size(range), true);
+            for(size_t i = 0; auto value : range)
```

Comment:
std::size_t

Reply:
(at other places too)

Reply:
I noticed that the library used "naked" size_t in various places. Should we are it should be `std::size_t` everywhere (I would prefer that)

Reply:
Yes. I noticed too. One weird thing is in one place, when I replaced with std::size_t, behavior changed! (warnings about signeness appeared).

In any case, std:: should be required.